src/label/label-image.js

```diff
@@ -14,35 +17,31 @@ const copyFile = require('../utils/copy-file');
  * up by the caption. e.g. 0.2 means 20% of the icon will have the caption.
  * @returns a promise which resolves with the result of the CLI command.
  */
-function caption(input, output, label, gravity, proportionalSize) {
+async function caption(input, output, label, gravity, proportionalSize) {
   //  Get the image width.
-  return getImageWidth(input)
-    .then((width) => {
-      //  The height is a proportional amount of the of the width. This means
-      //  with a square image, a proportionalSize of 0.2 and a gravity of 'bottom',
-      //  the bottom 20% of the icon will contain the caption.
-      const height = width * proportionalSize;
+  const width = await getImageWidth(input);
+  //  The height is a proportional amount of the of the width. This means
+  //  with a square image, a proportionalSize of 0.2 and a gravity of 'bottom',
+  //  the bottom 20% of the icon will contain the caption.
+  const height = width * proportionalSize;
 
-      //  Important: the quotes around the colour must be double quoutes for the
-      //  command to work on Windows!
-      const command = `convert \
-        -background "rgba(0,0,0,0.5)" -fill white \
-        -gravity center -size ${width}x${height} \
-        caption:"${label}" \
-        ${input} +swap -gravity ${gravity} -composite ${output}`;
-      return imagemagickCli.exec(command);
-    });
+  //  Important: the quotes around the colour must be double quoutes for the
+  //  command to work on Windows!
+  const command = `convert \
+    -background "rgba(0,0,0,0.5)" -fill white \
+    -gravity center -size ${width}x${height} \
+    caption:"${label}" \
+    ${input} +swap -gravity ${gravity} -composite ${output}`;
+  return imagemagickCli.exec(command);
 }
 
 //  Single function to label an image (optionally top and bottom).
-module.exports = function labelImage(input, output, top, bottom, middle) {
+module.exports = async function labelImage(input, output, top, bottom, middle) {
   //  First, create the output image. This will overwrite any existing file.
-  let promise = copyFile(input, output);
+  await copyFileAsync(input, output);
 
   //  We'll have a set of promises which we will run, which will update the image.
-  if (top) promise = promise.then(() => caption(output, output, top, 'north', 0.2));
-  if (bottom) promise = promise.then(() => caption(output, output, bottom, 'south', 0.2));
-  if (middle) promise = promise.then(() => caption(output, output, middle, 'center', 0.6));
-
-  return promise;
+  if (top) await caption(output, output, top, 'north', 0.2);
```

Comment:
Phew, it was so ugly before - again a recent change to support more labelling options which I knew had a limited shelf life due to `await/async` but still glad to see the old code go